packages/language-server/src/plugins/typescript/CSSClassDefinitionLocator.ts

```diff
@@ -0,0 +1,168 @@
+import { Position, Range } from 'vscode-languageserver';
+import { SvelteDocumentSnapshot } from './DocumentSnapshot';
+import { Document } from '../../lib/documents';
+import { SvelteNode } from './svelte-ast-utils';
+export class CSSClassDefinitionLocator {
+    initialNodeAt: SvelteNode;
+    cssClassTerminators = ['', '{', '.', '>', '~', '>', '[', ':', '#', '+'];
+    constructor(
+        public tsDoc: SvelteDocumentSnapshot,
+        public position: Position,
+        public document: Document
+    ) {
+        this.initialNodeAt = this.tsDoc.svelteNodeAt(this.position) as SvelteNode;
+    }
+
+    getCSSClassDefinition() {
```

Comment:
I believe it could be more promising/faster to defer CSS parsing to `vscode-css-languageservice` which has a parse method. The AST that is returned doesn't have type definitions because it's private, but I'm honestly not worried too much about that since it's unlikely to change for this. That AST could be traversed to more easily find the correct node to link to.

Reply:
Looking into this again. Using svelteNodeAt is giving back a lot of good Svelte information that I wouldn't think the vscode-css-languageservice would know about. This method will tell us all the other variations of Svelte css such as whether it is a ConditionalExpression or Class Directive Format.

Reply:
I didn't mean the HTML part, I meant the CSS part. The code does string traversal of the CSS to find the matching definition, but it could be more promising to use the CSS AST output of `vscode-css-languageservice`

Reply:
Are you referring to the while loop in getStandardFormatClassName() ? This is actually not traversing css but the class attribute string itself to figure out which classname the position is at  

`class="test te|st1 bar foo"` so in this example the cursor is in the middle of test1 so that's what we're going to look for in the style section